src/components/side_panel/chart/building_blocks/generic_side_panel/config_panel.ts

```diff
@@ -129,9 +129,9 @@ export class GenericChartConfigPanel extends Component<Props, SpreadsheetChildEn
       this.dataSets.length
     );
     const colors = this.dataSets.map((ds) => colorGenerator.next());
-    this.dataSets = indexes.map((i) => ({
-      backgroundColor: colors[i],
-      ...this.dataSets[i],
+    this.dataSets = indexes.map((index, i) => ({
+      backgroundColor: colors[index],
+      ...this.dataSets[index],
```

Comment:
useless change (and `i` is unused is you really want to change the variable name)